daemon/logger/fluentd/fluentd.go

```diff
@@ -84,6 +89,25 @@ func New(info logger.Info) (logger.Logger, error) {
 		return nil, err
 	}
 
+	writeTimeout := defaultWriteTimeout
+	i, _ := strconv.Atoi(info.Config[writeTimeoutKey])
+	if info.Config[writeTimeoutKey] != "" && i >= 0 {
+		wtd, err := time.ParseDuration(info.Config[writeTimeoutKey])
+		if err != nil {
+			statusCode := httputils.GetHTTPErrorStatusCode(err)
+			//Checks for 5xx status code error
+			switch statusCode {
+			case 501, 502, 503:
+				return nil, fmt.Errorf("Something went wrong please try again")
```

Comment:
I'm not sure I understand this error handling. `time.ParseDuration` shouldn't be making an HTTP request.

Reply:
yup it's wrong I realized later 🙈 .

Reply:
I am bit confused with the http status code. Where should I be checking it.

Reply:
Probably https://github.com/moby/moby/pull/35716/files#diff-4e76640044eca60a4a45cd8631049282R108 needs a properly typed error (of type `InvalidParameter`)

Reply:
How about ``` InvalidParameter: write-timeout must be a positive integer ``` @cpuguy83

Reply:
It's not the error message that matters so much as the error type, which needs to implement this interface: https://github.com/moby/moby/blob/master/api/errdefs/defs.go#L9